caddyhttp/httpserver/context.go

```diff
@@ -84,6 +85,30 @@ func (c Context) IP() string {
 	return ip
 }
 
+// ServerIP gets the (local) IP address of the server.
+func (c Context) ServerIP() string {
+	if c.NIFn == nil {
+		c.NIFn = net.InterfaceAddrs
+	}
+
+	addrs, err := c.NIFn()
+
+	if err != nil {
```

Comment:
This is a nit-pick, but conventionally in Go, you'll do the error check on the line immediately after calling your function, without an empty line in between.